packages/lit-dev-content/site/articles/article/lit-cheat-sheet.md

```diff
@@ -0,0 +1,1118 @@
+### Rendering lists
+
+Lit-html can render JavaScript arrays and iterables. For most simple use cases,
+you can use the `Array.map()` method to render arrays of items or the `map()`
+directive to render any other iterables. This pattern is best used for short,
+simple lists.
+
+{% playground-ide "articles/lit-cheat-sheet/render-lists", true %}
+
+**Related Documentation & Topics:**
+
+- [Lists](/docs/templates/lists/)
+- [Working With Lists Tutorial](/tutorials/working-with-lists/)
+- [Built-in directives – `map()`](/docs/templates/directives/#map)
+- [Built-in directives – `range()`](/docs/templates/directives/#range)
+- [Built-in directives – `join()`](/docs/templates/directives/#join)
+
+### Re-rendering lists efficiently
+
+For long lists that may change frequently, use the `repeat()` directive to efficiently re-render only the items that have changed.
+
+{% playground-ide "articles/lit-cheat-sheet/repeat", true %}
+
+- [Lists](/docs/templates/lists/)
+- [Working With Lists Tutorial – The `repeat()` directive](/tutorials/working-with-lists/#6)
+- [Built-in directives – `repeat()`](/docs/templates/directives/#repeat)
+
+### Virtualizing long lists
+
+For lists that are so long that it would be impractical to render all items at once, use the Lit Virtualizer to render only the items that are currently in view.
+
+{% playground-ide "articles/lit-cheat-sheet/virtualizer", true %}
+
+{% aside "labs" %}
+
+Lit Virtualizer is in labs
+
+meaning that its implementation might change until it has graduated and become stable. Additionally there are many more features to virtualizer, so it is recommended to look at the documentation.
+
+{% endaside %}
+
+**Related Documentation & Topics:**
+
+- [Lit Virtualizer Documentation](https://github.com/lit/lit/tree/main/packages/labs/virtualizer#readme) (external)
+- [Lit Labs: Virtualizer](https://www.youtube.com/watch?v=ay8ImAgO9ik) (video)
+
+### Render String to HTML
+
+To render a string of HTML as HTML in Lit, use the `unsafeHTML` directive.
+
+{% playground-ide "articles/lit-cheat-sheet/unsafe-html", true %}
+
+{% aside "negative" %}
+
+Be careful when using `unsafeHTML` as it can open your application up to
+cross-site scripting (XSS) and other attacks.
+
+Only use `unsafeHTML` with trusted sources and strings as you would use
+`Element.prototype.innerHTML`.
+
+{% endaside %}
+
+**Related Documentation & Topics:**
+
+- [Built-in directives – `unsafeHTML`](/docs/templates/directives/#unsafehtml)
+- [innerHTML Security considerations](https://developer.mozilla.org/en-US/docs/Web/API/Element/innerHTML#security_considerations) (external)
+
+### Bind to HTML Tag Name
+
+In rare cases, you need to bind to an HTML tag name to change the rendered
+element. You can do this safely with the `static-html` module and the `literal`
+template tag.
+
+{% playground-ide "articles/lit-cheat-sheet/bind-tag-name", true %}
+
+
+{% aside "warn" %}
+
+Be careful when using static HTML templates as changing values of a static
```

Comment:
They are cached but the caching is slightly more expensive. Instead I would just draw attention to the cost of changing element tags.